src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -604,6 +610,12 @@ private void ProcessSettingsFrame(FrameHeader frameHeader)
                     }
                 }
 
+                if (initialFrame && !maxConcurrentStreamsReceived)
+                {
+                    // Set to 'infinite' because MaxConcurrentStreams was not set on the initial SETTINGS frame.
+                    _maxConcurrentStreams = int.MaxValue;
```

Comment:
Should the call here be to ChangeMaxConcurrentStreams? That is what happens when `SettingId.MaxConcurrentStreams` setting is received.

Reply:
Yes, it should. Thanks for catching that.

Reply:
Fixed.